server.go

```diff
@@ -407,3 +422,45 @@ func (m *magicCheck) Read(buf []byte) (n int, err error) {
 	}
 	return n, err
 }
+
+type hostOption struct {
+	Name string
+	Host string
+	Port string
+}
+
+func parseHostOptions(options []string) []hostOption {
+	result := make([]hostOption, 0)
+
+	for _, row := range options {
+		option := strings.SplitN(row, ":", 3)
```

Comment:
Can this be changed to a `|` so it doesn't break IPv6 addresses?

Reply:
Alternatively, you could put the host as the last field, or find the first and last indexes of `:` and split on those instead.